crypto-extensions/src/main/java/net/corda/v5/crypto/extensions/DigestAlgorithm.java

```diff
@@ -0,0 +1,77 @@
+package net.corda.v5.crypto.extensions;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.InputStream;
+
+/**
+ * Digest extensions: Interface defining a custom digest calculation implementation. The interface should be implemented
+ * if a CPK developer wishes to provide support for digest algorithms beyond those supported by the Corda platform.
+ * The implementation of the interface must be coupled with the implementation of the [DigestAlgorithmFactory] which
+ * will be used to create the instances. For each algorithm there must be matching a pair of
+ * [DigestAlgorithmFactory] and [DigestAlgorithm] implementations.
+ * <p>
+ * Example of the algorithm and the factory implementations:
+ * <p>
+ * ```kotlin
```

Comment:
Javadoc does not support markdown. You will need to wrap this code block inside:
```
<pre>{@code .... }</pre>
```
Don't forget to reapply formatting :wink:.

These comments apply throughout.

Reply:
fixed in local copy

Reply:
pushed